oeps/oep-00XX-git-best-practices.rst

```diff
@@ -0,0 +1,107 @@
++-----------------+--------------------------------------------------------+
+| Last Modified   | 2019-07-18                                             |
++-----------------+--------------------------------------------------------+
+| Authors         | Joseph Mulloy                                          |
+|                 |                                                        |
++-----------------+--------------------------------------------------------+
+| Arbiter         | <Arbiter's real name and email address>                |
++-----------------+--------------------------------------------------------+
+| Status          | Draft                                                  |
++-----------------+--------------------------------------------------------+
+| Type            | Best Practice                                          |
++-----------------+--------------------------------------------------------+
+| Created         | 2019-07-18                                             |
++-----------------+--------------------------------------------------------+
+| `Review Period` | <start - target end dates for review>                  |
++-----------------+--------------------------------------------------------+
+| `Resolution`    | <links to any discussions where the final              |
+|                 | status was decided>                                    |
++-----------------+--------------------------------------------------------+
+| `References`    | <links to any other relevant discussions               |
+|                 | or relevant related materials>                         |
++-----------------+--------------------------------------------------------+
+
+Abstract
+========
+
+This OEP outlines git best practices that if followed increase the usability
+of the git commit history.
+
+Motivation
+==========
+
+Git history is often used to understand the motivations and context in which a
+change was made. When commit messages are descriptive and link to a ticket it's
+possible to understand why a change was made. When the message is terse it's
+much more difficult to understand the change.
+
+Specification
+=============
+
+The specification describes the technical details of the Architecture, Best
+Practice or Process proposed by the OEP. If the proposal includes a new API,
+specify its syntax and semantics.
+
+Commit messages
+---------------
+* The first line should be no longer than 50 characters
+* The second line should be no longer than 72 characters
+* The first line should concisely describe the reason for the change
+* The first line should include the JIRA Issue ID, if there is a JIRA Issue
+Branch names
+------------
+* Feature branch names should be prefixed with the creating user's github username and a slash
+* If there is an associated JIRA Issue the Issue ID should be next before another slash
+* Example: username/proj-1234/descriptive name
```

Comment:
Does this really matter? Don't we immediately delete the branches after merging anyhow?

Reply:
Fair point. It's something I do and I find it helpful. It is a bit redundant once there's a commit on the branch with the JIRA ID in the message. This could be a suggestion vs a requirement. Do OEPs have the shall and may language that RFCs have?

Reply:
By default, GitHub puts the branch name as the first line of any merge commit ([example](https://github.com/edx/edx-platform/commit/af7ee42bdefc9646b19ed9df0ad52ed29503f085)), so a clear branch name makes it easier to scan the list of merge commits to see what each merge was. I wish they would change to put the PR title as the first line, which is much more useful.

I personally like to override the GitHub default and change the merge commit first line to
"Merge PR #NNNNN: Title of the PR"
vs.
"Merge pull request #NNNNN from username/partial-branch-nam..."

Reply:
I think it's a useful convention, but doesn't belong in an OEP.